lrpar/src/lib/ctbuilder.rs

```diff
@@ -73,14 +74,58 @@ pub enum ActionKind {
     GenericParseTree
 }
 
+struct CTConflictsError<StorageT: Eq + Hash> {
+    pub grm: YaccGrammar<StorageT>,
+    pub sgraph: StateGraph<StorageT>,
+    pub stable: StateTable<StorageT>
+}
+
+impl<StorageT> fmt::Display for CTConflictsError<StorageT>
+where
+    StorageT: 'static + Debug + Hash + PrimInt + Serialize + TypeName + Unsigned,
+    usize: AsPrimitive<StorageT>,
+    u32: AsPrimitive<StorageT>
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let conflicts = self.stable.conflicts().unwrap();
+        write!(f, "{}", conflicts.pp(&self.grm)) //
+    }
+}
+
+impl<StorageT> fmt::Debug for CTConflictsError<StorageT>
```

Comment:
Can we not `#[derive(Debug)]` something that's good enough for debug support?

Reply:
Then we would also have to add `#[derive(Debug)]` to `YaccGrammar` and `StateTable`, and I'm not sure what the output would then be.

Reply:
We have to implement `Debug` because of `Error` I guess?

Reply:
I'm not sure what you mean. How I see it we have 3 options:
1) We use `#[derive(Debug)]`, which means we also have to add to `YaccGrammar` and `StateTable`, and when this error is returned it prints out the entire grammar and statetable. Which IMO is a bit too much information.

2) We leave this is at is, printing out something like `CTConflictsError(1 Shift/Reduce, 5 Reduce/Reduce)` as a debug message.

3) We update this to print out the stategraph + the actual errors encountered, resulting in something like:
```
0:  [^ -> . Expr, {'$'}]
    Term -> 2
    Factor -> 4
    Expr -> 3
    'INT' -> 5
    'LBRACK' -> 1
1:  [Factor -> 'LBRACK' . Expr 'RBRACK', {'PLUS', 'MUL', 'RBRACK', '$'}]
    Expr -> 6
    'LBRACK' -> 1
    'INT' -> 5
    Factor -> 4
    Term -> 2
...

Shift/Reduce conflicts:
   State 2: Shift("PLUS") / Reduce(Term: "Term")
Reduce/Reduce conflicts:
   State 2: Reduce(Expr: "Term") / Reduce(Term: "Term")
   State 2: Reduce(Expr: "Term") / Reduce(Term: "Term")
   State 11: Reduce(Term: "Factor" "MUL" "Term") / Reduce(Term: "Term")
   State 11: Reduce(Term: "Factor" "MUL" "Term") / Reduce(Term: "Term")
   State 11: Reduce(Term: "Factor" "MUL" "Term") / Reduce(Term: "Term")

```

I'm probably tending to the last one, since then when the user sets `error_on_conflicts(true)`, they immediately get all the info they need to debug this error. Otherwise they have to first set `error_on_conflicts(false)` and then retrieve the `ConflictsStruct` from the result, and then pretty print it using a `panic!`.

Reply:
I guess what I'm asking is: do we need to implement `Debug` at all?

Reply:
It seems it's needed by `Error`:
```
impl<StorageT> Error for CTConflictsError<StorageT>                                                                                                                                     
    |                ^^^^^ `ctbuilder::CTConflictsError<StorageT>` cannot be formatted using `{:?}` 
```

Reply:
Right, OK, that was my guess. Then I'm more-or-less OK with this as-is. Should it be `CTConflictsError{{{} Shift/Reduce, {} Reduce/Reduce}}` to make it look more like other debug output?

Reply:
Can do. What's your thoughts on my suggestion above to output the graph and more detailed conflicts instead, so the user doesn't need to jump through so many hoops to get to this information.

Reply:
I think `Debug` output should me minimal, so I'd prefer not to output the graph etc.

Reply:
Yes, I agree `Debug` should be minimal. I've moved the pretty printed stuff into the `Display` method, though I'm not sure if this is the correct place for it either.

Reply:
I must admit, I generally make `Debug` and `Display` the same; and if someone wants the full pretty-printed output I'd make them call `pp`. But this is something over which reasonable people differ.

Reply:
Okay, looks go with your way then to be consistent.

Reply:
If they want a prettyprinted version for the error, they can just call `error.cs.pp()`.

Reply:
Works for me.